### PR TITLE
Enable TenantId to Retrieve Auxiliary Auth Header

### PR DESCRIPTION
**Pull request by msJinLei**
This pull request includes changes to add new unit tests and methods related to authentication functionality. The most important changes include adding a new unit test class `AzureRMCmdletAuthenticationUnitTests` with test methods for the `GetAuxilaryAuthHeaderByTenatIds` method, and adding a new method `GetAuxiliaryAuthHeaderFromTenantIds` to the `AzureRMCmdlet` class for generating authentication headers.

Authentication improvements:

* [`src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs`](diffhunk://#diff-4fc6524a8cc5579c0ba48a332feff18740fd36a7976cf50ff058859f316044b2R150-R174): Added a new method `GetAuxiliaryAuthHeaderFromTenantIds` to the `AzureRMCmdlet` class for generating authentication headers for a list of tenant IDs.

Testing improvements:

* [`src/ResourceManager.Test/AzureRMCmdletAuthenticationUnitTests.cs`](diffhunk://#diff-7612c52daee7f43404df8e6977910156cc47a998893972bc5702d6c5968308cbR1-R159): Added a new unit test class `AzureRMCmdletAuthenticationUnitTests` with test methods to test the functionality of the `GetAuxilaryAuthHeaderByTenatIds` method.

### CODE REVIEW

File: src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs [hunk 1]
```diff
@@ -147,6 +147,31 @@ private bool ShouldCloneDefaultProfile()
 
         private IAzureContextContainer _clonedDefaultProfile;
 
+        protected IDictionary<string, List<string>> GetAuxiliaryAuthHeaderFromTenantIds(IEnumerable<string> tenantIds)
```
Comment (isra-fel) [inline src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs:150]: IList?

File: src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs [hunk 2]
```diff
@@ -147,6 +147,31 @@ private bool ShouldCloneDefaultProfile()
...
+        {
+            if ((tenantIds != null) && (tenantIds.Count() > 0))
+            {
+                // WE can only fill in tokens for 3 tennats in the aux header, if tehre are more tenants fail now
+                if (tenantIds.Count() > MAX_NUMBER_OF_TOKENS_ALLOWED_IN_AUX_HEADER)
+                {
+                    throw new ArgumentException("Number of tenants (tenants other than the one in the current context), that the requested resources belongs to, exceeds maximum allowed number of " + MAX_NUMBER_OF_TOKENS_ALLOWED_IN_AUX_HEADER);
```
Comment (isra-fel) [inline src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs:157]:
```suggestion
                    throw new ArgumentException("Number of tenants (tenants other than the one in the current context), that the requested resources belong to, exceeds maximum allowed number of " + MAX_NUMBER_OF_TOKENS_ALLOWED_IN_AUX_HEADER);
```

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs [hunk 3]
```diff
@@ -147,6 +147,31 @@ private bool ShouldCloneDefaultProfile()
...
+                }
+
+                //get the tokens for each tenant and prepare the string in the following format :
+                //"Header Value :: Bearer <auxiliary token1>;EncryptedBearer <auxiliary token2>; Bearer <auxiliary token3>"
+
+                var tokens = tenantIds.Select(t => (new StringBuilder(AUX_TOKEN_PREFIX).Append(" ").Append(GetTokenForTenant(t)?.AccessToken))?.ToString())?.ConcatStrings(AUX_TOKEN_APPEND_CHAR);
```
Comment (wyunchi-ms) [inline src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs:163]:
```suggestion
var tokens = tenantIds
    .Select(t => $"{AUX_TOKEN_PREFIX} {GetTokenForTenant(t)?.AccessToken}")
    .ConcatStrings(AUX_TOKEN_APPEND_CHAR);
```

**Review by isra-fel — APPROVED** [review]
Please provide context in PR descrption. You can leverage the AIG draft BTW

**Review by wyunchi-ms — APPROVED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**